homework/index.js

```diff
@@ -30,17 +15,201 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function sortElements(repositories) {
+    return repositories.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Hi Yash, thank you for this information, it is very useful. But if I delete the `return` here, the code gives an error like this: 
`index.js:146 Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined
    at createHtml (index.js:146)
    at main (index.js:204)
createHtml @ index.js:146
main @ index.js:204
async function (async)
main @ index.js:201
window.onload @ index.js:215
load (async)
(anonymous) @ index.js:215`

But `[...repositories]` work with `return` perfectly.

Reply:
Hi Ahmet, I think I wasn't explicit in my earlier comment but I believe that what you tried was:
```
  function sortElements(repositories) {
    repositories.sort((a, b) => a.name.localeCompare(b.name));
  }

  // Inside createHtml
  const sortedRepositories = sortElements(repositories);
```

If this is what you tried then you would face the above error. This is because the function `sortElements` returns `undefined` implicitly. You can simply comment out the line in `createHtml` method and use `repositories` variable instead. Can you try that?

Reply:
Hi Yash, now I just try to invoke `sortElements` method, I deleted the assignment of the returned value of the `sortElements` function, I changed `sortedRepositories` variable to `repositories` and as you said it works. Thanks for the information.